clippy_lints/src/methods/unwrap_or_else_default.rs

```diff
@@ -23,6 +23,25 @@ pub(super) fn check<'tcx>(
     let is_option = is_type_diagnostic_item(cx, recv_ty, sym::Option);
     let is_result = is_type_diagnostic_item(cx, recv_ty, sym::Result);
 
+    if let hir::ExprKind::Closure(&hir::Closure { body, .. }) = u_arg.kind {
+        let body = cx.tcx.hir().body(body);
+        if closure_body_returns_empty_string(body) {
+            let mut applicability = Applicability::MachineApplicable;
+            span_lint_and_sugg(
+                cx,
+                UNWRAP_OR_ELSE_DEFAULT,
+                expr.span,
+                "use of `.unwrap_or_else(..)` to construct default value",
+                "try",
+                format!(
+                    "{}.unwrap_or_default()",
+                    snippet_with_applicability(cx, recv.span, "..", &mut applicability)
+                ),
+                applicability,
+            );
```

Comment:
This is the same as below, so can this check be put into the following?

Reply:
but I have to do this https://github.com/rust-lang/rust-clippy/compare/a07f9ccc4244566ba1a60c04d653862f891e07e8..0160e12edd98423a3b1b69e07688930d57ac5aab#diff-eee44b9635bbab1626371d693f19dd5366810f29fa80e83f6e053c368ef09785R35-R36

`if_chain!` becomes suddenly not easily extensible anymore.